src/Features/Core/Portable/DisposeAnalysis/DisposableFieldsShouldBeDisposedDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,266 @@
+                // and were assigned a disposable object within this containing type, but were not disposed in
+                // containing type's Dispose method.
+                symbolStartContext.RegisterSymbolEndAction(analyzer.OnSymbolEnd);
+            }
+
+            private void AddOrUpdateFieldDisposedValue(IFieldSymbol field, bool disposed)
+            {
+                Debug.Assert(_disposableFields.Contains(field));
+                Debug.Assert(!field.IsStatic);
+                Debug.Assert(field.Type.IsDisposable(_disposeAnalysisHelper.IDisposable));
+
+                // Update the dispose value for the field.
+                // Update value factory delegate ensures that fields for which we have
+                // already seen dispose invocations, i.e. currentValue = true, continue to be marked as disposed.
+                _fieldDisposeValueMap.AddOrUpdate(field,
+                    addValue: disposed,
+                    updateValueFactory: (f, currentValue) => currentValue || disposed);
+            }
+
+            private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
+            {
+                foreach (var kvp in _fieldDisposeValueMap)
+                {
+                    IFieldSymbol field = kvp.Key;
+                    bool disposed = kvp.Value;
+                    if (!disposed)
+                    {
+                        // Disposable field '{0}' is never disposed
+                        var diagnostic = Diagnostic.Create(s_disposableFieldsShouldBeDisposedRule, field.Locations[0], field.Name);
+                        symbolEndContext.ReportDiagnostic(diagnostic);
+                    }
+                }
+            }
+
+            private void OnOperationBlockStart(OperationBlockStartAnalysisContext operationBlockStartContext)
+            {
+                switch (operationBlockStartContext.OwningSymbol)
+                {
+                    case IFieldSymbol _:
+                        // Field initializer.
+                        if (operationBlockStartContext.OperationBlocks.Length == 1 &&
+                            operationBlockStartContext.OperationBlocks[0] is IFieldInitializerOperation fieldInitializer)
+                        {
+                            foreach (var field in fieldInitializer.InitializedFields)
+                            {
+                                if (!field.IsStatic && _disposableFields.Contains(field))
+                                {
+                                    // Instance field initialized with a disposable object is considered a candidate.
+                                    AddOrUpdateFieldDisposedValue(field, disposed: false);
+                                }
+                            }
+                        }
+
+                        break;
+
+                    case IMethodSymbol containingMethod:
```

Comment:
I guess property or event get/set/add/remove is at the end IMethodSymbol and called here.

Reply:
Yes, that is correct.